Kernel/Syscall.cpp

```diff
@@ -108,8 +108,25 @@ KResultOr<FlatPtr> handle(RegisterState& regs, FlatPtr function, FlatPtr arg1, F
     auto& process = current_thread->process();
     current_thread->did_syscall();
 
+    if (function >= Function::__Count) {
+        dbgln("Unknown syscall {} requested ({:08x}, {:08x}, {:08x})", function, arg1, arg2, arg3);
+        return ENOSYS;
+    }
+
+    const auto syscall_metadata = s_syscall_table[function];
+    if (syscall_metadata.handler == nullptr) {
+        dbgln("Null syscall {} requested, you probably need to rebuild this program!", function);
+        return ENOSYS;
+    }
+
+    const auto needs_big_lock = syscall_metadata.needs_lock == NeedsBigProcessLock::Yes;
+    if (needs_big_lock) {
+        process.big_lock().lock();
+    }
```

Comment:
Maybe use a `ScopeGuard` for the unlock to make sure we don't accidentally return with the lock held. It's not a problem right now but might be in the future if someone adds another `return` statement. :)

Reply:
Sure, I've now modified `MutexLocker` to allow it to be conditionally initialized.